lib/src/Pos/Web/Server.hs

```diff
@@ -58,10 +56,26 @@ type MyWorkMode ctx m =
     , Default (MempoolExt m)
     )
 
-route53HealthCheckApplication :: MyWorkMode ctx m => Topology t -> OQ m -> m Application
-route53HealthCheckApplication topology oq = do
-    server <- servantServerHealthCheck topology oq
-    return $ serve healthCheckApi server
+withRoute53HealthCheckApplication
+    :: ( Mockable Async m
+       , MonadMask m
+       , MonadIO m
+       , HasConfiguration
+       )
+    => IO HealthStatus
+    -> String
+    -> Word16
+    -> m x
+    -> m x
+withRoute53HealthCheckApplication mStatus host port act = bracket acquire release (const act)
```

Comment:
Why are you using `bracket async cancel` instead of `withAsync`? I know they are different, by default I would stick to `withAsync`. Is there an important reason why `withAsync` can't be used here?

Reply:
No it can be `withAsync`. I'll make the change.